### ISSUE TITLE
[REVIEW]: sluRm: A lightweight wrapper for HPC with Slurm

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1493]
**Submitting author:** @gvegayon (<a href="http://orcid.org/0000-0002-3171-0844">George Vega Yon</a>)
**Repository:** <a href="https://github.com/USCbiostats/sluRm" target ="_blank">https://github.com/USCbiostats/sluRm</a>
**Version:** v0.2-0
**Editor:** @karthik
**Reviewer:** @mschubert, @mllg
**Archive:** <a href="https://doi.org/10.5281/zenodo.3272773" target="_blank">10.5281/zenodo.3272773</a>

## Status

[![status](http://joss.theoj.org/papers/5e1564ca907e4b47accc722ab2913826/status.svg)](http://joss.theoj.org/papers/5e1564ca907e4b47accc722ab2913826)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/5e1564ca907e4b47accc722ab2913826"><img src="http://joss.theoj.org/papers/5e1564ca907e4b47accc722ab2913826/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/5e1564ca907e4b47accc722ab2913826/status.svg)](http://joss.theoj.org/papers/5e1564ca907e4b47accc722ab2913826)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/USCbiostats/sluRm" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@mschubert & @mllg, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @karthik know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/USCbiostats/sluRm">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v0.2-0
- [x] **Authorship:** Has the submitting author (@gvegayon) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/USCbiostats/sluRm">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** v0.2-0
- [x] **Authorship:** Has the submitting author (@gvegayon) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @mschubert, @mllg it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01493/joss.01493/10.21105.joss.01493.pdf)

**Comment by karthik** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- None

MISSING DOIs

- https://doi.org/10.1007/10968987_3 may be missing for title: SLURM: Simple Linux Utility for Resource Management

INVALID DOIs

- None
```

**Comment by mschubert** [comment]
I finished my first 2 passes. Nice work @gvegayon and @pmarjora!

I'm happy for this to be accepted if the following issues are addressed: [#6](https://github.com/USCbiostats/sluRm/issues/6), [#7](https://github.com/USCbiostats/sluRm/issues/7), [#8](https://github.com/USCbiostats/sluRm/issues/8), [#13](https://github.com/USCbiostats/sluRm/issues/13) (others are optional, but would be nice to have)

**Comment by gvegayon** [comment]
@mschubert this is great, we really appreciate it! We will address those (and the optional issues) ASAP. Thanks!

**Comment by mllg** [comment]
I've also finished my first pass. Issues are referenced here, #14 and #15 are mandatory.

**Comment by karthik** [comment]
@gvegayon Please update us once you've addressed these issues.

**Comment by gvegayon** [comment]
OK @mschubert @mllg, I've worked on the issues you created on the project's repo. Addressed all of the required ones plus some of those which were not mandatory (thanks!). LMK if you are OK with the changes, and if so, please close the issues that you consider complete.

Also, following recent changes on how to list contributions on R packages (if you are OK with it) I've added both of you in the list of authors with the role of `"rev"` (see [here](https://github.com/USCbiostats/sluRm/blob/91909a7c4fd4b94811d473384b56712285e7baa5/DESCRIPTION#L8-L9)).

PS: A thing that I'm not sure whether JOSS does but I think is worthy is to keep track of how much projects change after submission. I am proud to say that my `git diff --stat` is (+)917 (-)269 lines since the package was submitted (on R/Rmd/md files), so yes, this review has required me to make significant changes towards I think is an improved version of the package, in other words, I think the review process works!

**Comment by karthik** [comment]
Thanks @gvegayon! I'm glad you added the reviewers with the appropriate MARC roles (it's recommended at rOpenSci but not here). If either reviewer objects or declines, you can revert the change. 

Glad to see the process working! Please see another issue I raised on your repo.

**Comment by karthik** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by karthik** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01493/joss.01493/10.21105.joss.01493.pdf)

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1007/10968987_3 is OK
- 10.21105/joss.00135 is OK
- 10.1093/bioinformatics/btz284 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by karthik** [comment]
Everything LGTM.

Over you to @openjournals/joss-eics

**Comment by mschubert** [comment]
`drake` citation looks wrong, shouldn't this be JOSS instead of the package URL?

**Comment by kyleniemeyer** [comment]
Yes, @gvegayon, please update the drake citation to the JOSS paper: 
```
@article{MichaelLandau2018,
  doi = {10.21105/joss.00550},
  year = {2018},
  month = jan,
  volume = {3},
  number = {21},
  pages = {550},
  author = {William Michael Landau},
  title = {The drake {R} package: a pipeline toolkit for reproducibility and high-performance computing},
  journal = {The Journal of Open Source Software}
}```

**Comment by gvegayon** [comment]
> `drake` citation looks wrong, shouldn't this be JOSS instead of the package URL?

Yes @mschubert, thanks @kyleniemeyer, I looked for papers related to drake on google scholar without success (perhaps needed to look harder?). Changed.

@karthik I see on the bottom of the pdf version is skipping part of my last name (is *Vega Yon*, not just *Yon*). I know this may be a bit painful, but I was wondering if there's any way to fix that? If not, perhaps a PR to JOSS to follow an approach as the `person(given=, family=,middle=)` function in R (for the future, of course).

**Comment by kyleniemeyer** [comment]
@gvegayon oh, good catch on your name. 

@arfon, is there a way to ensure whedon captures compound / double surnames? Would wrapping it in `{ }` work, as in BibTeX?

**Comment by kyleniemeyer** [comment]
Update: using a local compile script, I tried both `{ }` and `" "`, and neither worked. This might require a local paper build by @arfon for now...

**Comment by karthik** [comment]
Thanks @kyleniemeyer and @gvegayon for catching this. Let's wait for @arfon to fix this manually.

**Comment by arfon** [comment]
> Thanks @kyleniemeyer and @gvegayon for catching this. Let's wait for @arfon to fix this manually.

Yes, this will have to be fixed manually at the end of the review. From a quick inspection it looks like we're not done with this review yet? If so, I'll fix this once the paper is ready to be accepted.

**Comment by kyleniemeyer** [comment]
Ah, I didn't notice the checklists.

@mschubert and @mllg, can you confirm that all the outstanding checklist items are satisfied? (And check them off if so)

**Comment by mschubert** [comment]
Looks good to me!

I haven't seen any comments from @pmarjora, is it customary for JOSS that he also confirms his authorship?

Last minor comments: Typos in the paper (resubmission, "the later" should be "the latter") and README ("Whereas is reporting" doesn't make sense, did you mean "Whether it is"?)

**Comment by karthik** [comment]
> I haven't seen any comments from @pmarjora, is it customary for JOSS that he also confirms his authorship?

We don't require every author to confirm separately.

**Comment by mllg** [comment]
The issues I raised have been adequately addressed. From my side, this is ready for publication.

**Comment by karthik** [comment]
Looks like both reviewers have now properly signed off. Over to you again @openjournals/joss-eics

**Comment by labarba** [comment]
👋  @gvegayon — The JOSS publication process requires that after all revisions to the software and paper are complete and the reviewers and editor recommend publication, you do the following:

- make a tagged release of your software, and report the version number here so we can update the metadata
- make a full deposit of your software in an archive that provides persistence and a global identifier such as a DOI (Zenodo is the most popular, but other archives with similar services exist and are permitted); report the DOI of the software archive here, so we can update the metadata.

You may find this helpful: https://guides.github.com/activities/citable-code/

**Comment by gvegayon** [comment]
Hey @labarba , the DOI is 10.5281/zenodo.3272773 Thanks!

**Comment by labarba** [comment]
Please report the version number here, so we can updated the metadata.

**Comment by labarba** [comment]
And the DOI unfortunately does not resolve http://doi.org/10.5281/zenodo.3272773

**Comment by labarba** [comment]
I found the tag of the version https://github.com/USCbiostats/sluRm/releases

**Comment by labarba** [comment]
@whedon set v0.2-0 as version

**Comment by whedon** [comment]
OK. v0.2-0 is the version.

**Comment by gvegayon** [comment]
> And the DOI unfortunately does not resolve http://doi.org/10.5281/zenodo.3272773

Now it does :)

**Comment by labarba** [comment]
Please check the Zenodo deposit has the correct metadata, this includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it); you may also add the authors' ORCID.

**Comment by gvegayon** [comment]
> Please check the Zenodo deposit has the correct metadata, this includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it); you may also add the authors' ORCID.

Just did: Authors (with ORCID), title and description (as provided in the R package). Question: On the **upload type** field, should I check **Publication** or **Software**?

**Comment by labarba** [comment]
Software. The Zenodo deposit is an archive of the software.

**Comment by labarba** [comment]
@whedon set 10.5281/zenodo.3272773 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3272773" target="_blank">10.5281/zenodo.3272773</a> is the archive.

**Comment by labarba** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1007/10968987_3 is OK
- 10.21105/joss.00135 is OK
- 10.1093/bioinformatics/btz284 is OK
- 10.21105/joss.00550 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/826

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/826, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by labarba** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1148352222293626880) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/827
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01493](https://doi.org/10.21105/joss.01493)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by labarba** [comment]
Congratulations, @gvegayon, your JOSS paper is published! 🚀 

Big thanks to our editor: @karthik, and reviewers: @mschubert, @mllg —JOSS runs on your contribution 🙏

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.01493/status.svg)](https://doi.org/10.21105/joss.01493)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01493">
  <img src="http://joss.theoj.org/papers/10.21105/joss.01493/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: http://joss.theoj.org/papers/10.21105/joss.01493/status.svg
   :target: https://doi.org/10.21105/joss.01493
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.01493/status.svg)](https://doi.org/10.21105/joss.01493)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.salsalabs.org/donate-to-joss

**Comment by gvegayon** [comment]
Great! Thanks :)! one small problem! my last name is "Vega Yon", not "Yon" (see [here](https://github.com/openjournals/joss-reviews/issues/1493#issuecomment-505993965))

**Comment by labarba** [comment]
Ping @arfon for this ☝️

**Comment by arfon** [comment]
> Ping @arfon for this ☝️

This should be fixed now. The PDF on the JOSS site can take a while to refresh as it's cached.

**Comment by gvegayon** [comment]
Hi @arfon I've finished rebranding sluRm to slurmR (paper included). The version of the repository that should be published is here: https://github.com/USCbiostats/slurmR/tree/257577f2197747ea926dfc9a7349c795d7a85e88
Thanks!

**Comment by arfon** [comment]
> Hi @arfon I've finished rebranding sluRm to slurmR (paper included). The version of the repository that should be published is here: https://github.com/USCbiostats/slurmR/tree/257577f2197747ea926dfc9a7349c795d7a85e88

Roger that. The paper should now be updated although it might take a few hours to show up modified on the JOSS site as the PDFs are cached.

**Comment by gvegayon** [comment]
Hey @arfon! I realized that my [last name has not been fixed for the citation](https://github.com/openjournals/joss-reviews/issues/1493#issuecomment-509410314). No rush, but I would appreciate if you could take a look :D!

**Comment by arfon** [comment]
@gvegayon – OK, this should be updated now.
